paddle/fluid/imperative/partial_grad_engine.cc

```diff
@@ -36,6 +36,15 @@
 namespace paddle {
 namespace imperative {
 
+struct HashPair {
+  template <class T1, class T2>
+  size_t operator()(const std::pair<T1, T2> &p) const {
```

Comment:
Prefer to add the `noexcept` key word here for the consistency with stl hash, namely `size_t operator()(const std::pair<T1, T2> &p) const noexcept`

Reply:
DONE